packages/connection-storage/src/connection-storage.ts

```diff
@@ -213,6 +214,14 @@ export class ConnectionStorage {
       : undefined;
   }
 
+  private static migrateSavedConnectionType(
```

Comment:
When do these gets saved back to the storage layer?

Reply:
Sorry I am maybe missing memory on our conversation here when you explained this to me. But is that this migration is only for how Compass will use this information but on disk we will keep the data structure as it is?

Reply:
Yes, we will keep the data structure in disk as it is, we only change how we see it in Compass. Maurizio suggested a few changes on the storage layer, but we will do them in another tasks and it requires a bigger refactor.